### DISCUSSION TITLE
How to rebuild css or add extra css

### DISCUSSION THREAD

**Discussion by rabol** [filamentphp/filament#506] [Help]
Hi

I'm using a very simple APM package, and was trying to create a dashboard widget to show the stats, and I noticed that the css that is included in Filament does not include all colors etc.

I know that Tailwind purge unused stuff :)

But what is the best way to rebuild the css or include my own (Yes, I have seen the doc about registering my assets) but I was wondering how that would work in real life

eg. if I try to install tailwind css in my current project I get an error for the dependencies as Filament uses 'old' Tailwind

**Comment by ryangjchandler** [comment]
@rabol Would you be able to provide your Tailwind config, etc. I can't see how the dependencies would conflict with Filament unless you're compiling Filament's CSS alongside yours?

**Reply by rabol** [reply]
My is a bootstrap project, so before installing Filament I did not have tailwind, but simply issuing this gives an error:

```
npm install -D tailwindcss@latest postcss@latest autoprefixer@latest
```

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: tailwindcss@2.2.7
npm ERR! node_modules/tailwindcss
npm ERR!   dev tailwindcss@"2.2.7" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer tailwindcss@"^1.0" from @tailwindcss/custom-forms@0.2.1
npm ERR! node_modules/@tailwindcss/custom-forms
npm ERR!   dev @tailwindcss/custom-forms@"^0.2.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/vagrant/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/vagrant/.npm/_logs/2021-07-27T08_51_00_159Z-debug.log
```


Now... I tried this:

npx init
then create a /resource/css/tailwind.css

then in my tailwind css
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

and the I added this to my webpack.mix.js
```
.postCss("resources/css/tailwind.css", "public/css", [
        require("tailwindcss")

    ]);

```

then run npm dev

i then created a new service provider to include my new css

```
<?php

namespace App\Providers;

use Filament\PluginServiceProvider;
use Illuminate\Support\ServiceProvider;

class BackendServiceProvider extends PluginServiceProvider
{
    protected $styles = [
        'my-package-styles' => 'tailwind.css',
    ];

}
```

it get's executed, but the css is not included because it looks in /admin/assets/tailwind.css which does not exists

so I'm almost there :)

**Reply by rabol** [reply]
After reading the docs one more time:

```
protected function styles()
    {
        return [
            'my-package-styles' => asset('/css/tailwind.css'),
        ];
    }
```

But it is not the best solution as I can see that my version overwrite some of the Filament css

**Reply by ryangjchandler** [reply]
I'd recommend looking into something like: https://tailwindcss.com/docs/configuration#selector-strategy

**Reply by Milkhan** [reply]
Any update on this? Stuck in the same place.

**Reply by danharrin** [reply]
Use a custom theme - https://filamentphp.com/docs/2.x/admin/appearance#building-themes - and make sure to add your view paths to `content` in tailwind.config.

**Reply by Milkhan** [reply]
@danharrin You are a life saver! Thank you!
